molgenis-data-validation/src/main/java/org/molgenis/data/validation/meta/AttributeValidator.java

```diff
@@ -308,57 +309,87 @@ private static void validateOrderBy(Attribute attr, Sort orderBy)
 		// transitions to FILE not allowed because associated file in FileStore not created/removed
-		DATA_TYPE_DISALLOWED_TRANSITIONS = new EnumMap<>(AttributeType.class);
-		DATA_TYPE_DISALLOWED_TRANSITIONS
-				.put(BOOL, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, EMAIL, HYPERLINK, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS
-				.put(CATEGORICAL, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, EMAIL, HYPERLINK, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS.put(CATEGORICAL_MREF, EnumSet.complementOf(EnumSet.of(MREF)));
-		DATA_TYPE_DISALLOWED_TRANSITIONS
-				.put(COMPOUND, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, EMAIL, HYPERLINK, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS
-				.put(DATE, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, EMAIL, HYPERLINK, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS
-				.put(DATE_TIME, EnumSet.of(CATEGORICAL_MREF, ONE_TO_MANY, MREF, EMAIL, HYPERLINK, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS
-				.put(DECIMAL, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, EMAIL, HYPERLINK, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS.put(EMAIL, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, HYPERLINK, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS
-				.put(ENUM, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, EMAIL, HYPERLINK, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS.put(FILE, EnumSet.allOf(AttributeType.class));
-		DATA_TYPE_DISALLOWED_TRANSITIONS
-				.put(HTML, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, EMAIL, HYPERLINK, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS.put(HYPERLINK, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, EMAIL, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS
-				.put(INT, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, EMAIL, HYPERLINK, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS
-				.put(LONG, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, EMAIL, HYPERLINK, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS.put(MREF, EnumSet.complementOf(EnumSet.of(CATEGORICAL_MREF)));
-		DATA_TYPE_DISALLOWED_TRANSITIONS.put(ONE_TO_MANY, EnumSet.allOf(AttributeType.class));
-		DATA_TYPE_DISALLOWED_TRANSITIONS
-				.put(SCRIPT, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, EMAIL, HYPERLINK, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS
-				.put(STRING, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, EMAIL, HYPERLINK, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS
-				.put(TEXT, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, EMAIL, HYPERLINK, FILE));
-		DATA_TYPE_DISALLOWED_TRANSITIONS
-				.put(XREF, EnumSet.of(CATEGORICAL_MREF, MREF, ONE_TO_MANY, EMAIL, HYPERLINK, FILE));
+		DATA_TYPE_ALLOWED_TRANSITIONS = new EnumMap<>(AttributeType.class);
+		EnumSet<AttributeType> allowedIdAttributeTypes = getValidIdAttributeTypes();
+
+		// TRUE and FALSE can either be expressed in string or 0 and 1
+		// Postgres does not support boolean to bigint or double precision (LONG and DECIMAL)
+		DATA_TYPE_ALLOWED_TRANSITIONS.put(BOOL, EnumSet.of(STRING, TEXT, INT));
```

Comment:
BOOL to DECIMAL will throw an error, why support it?
BOOl to ENUM does not make sense, why would you create an ENUM for the two options BOOL already supports?
BOOL to LONG will throw an error, why support it?

To convert to HTML and SCRIPT just feels as if we are doing it because they are text fields. You might as well convert to STRING and then convert to HTML or SCRIPT